ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,568 @@
+)
+
+func ResourceIBMCOSBucketLifecycleConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceIBMCOSBucketLifecycleConfigurationCreate,
+		Read:          resourceIBMCOSBucketLifecycleConfigurationRead,
+		Update:        resourceIBMCOSBucketLifecycleConfigurationUpdate,
+		Delete:        resourceIBMCOSBucketLifecycleConfigurationDelete,
+		Importer:      &schema.ResourceImporter{},
+		CustomizeDiff: resourceValidateLifecycleRule,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"bucket_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket CRN",
+			},
+			"bucket_location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "COS bucket location",
+			},
+			"endpoint_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"public", "private", "direct"}),
+				Description:  "COS endpoint type: public, private, direct",
+				Default:      "public",
+			},
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"abort_incomplete_multipart_upload": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"days_after_initiation": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"expiration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validate.ValidateUTCFormat,
+									},
+									"days": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ValidateFunc: validate.ValidateAllowedRangeInt(1, 3650),
+										Default:      0, // API returns 0
+									},
+									"expired_object_delete_marker": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"filter": {
+							Type:             schema.TypeList,
+							Required:         true,
+							DiffSuppressFunc: suppressMissingFilterConfigurationBlock,
+							// IBM has filter a required parameter
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"prefix": {
+										Type:     schema.TypeString, // check if prefix empty is eccepted and if filter empty is accepted
+										Optional: true,
```

Comment:
shouldn't it be a required field i.e can i pass a filter like this filter{} without prefix?

Reply:
what about prefix is null case?

Reply:
filter is a required fields not prefix. empty prefix is accepted - prefix = "" , also something like filter {} is handled by the code itself. We have test cases for these 2 scenarios.